Example/src/screens/HeaderOptions.tsx

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable react/display-name */
+import React, {useState, useLayoutEffect} from 'react';
+import {
+  SafeAreaView,
+  StyleSheet,
+  ScrollView,
+  Text,
+  Platform,
+} from 'react-native';
+import {enableScreens} from 'react-native-screens';
+import {
+  createNativeStackNavigator,
+  NativeStackNavigationProp,
+} from 'react-native-screens/native-stack';
+
+import {
+  SettingsInput,
+  SettingsPicker,
+  SettingsSwitch,
+  Square,
+  Button,
+} from '../shared';
+
+enableScreens();
+
+type StackParamList = {
+  Main: undefined;
+  Settings: undefined;
+};
+
+interface MainScreenProps {
+  navigation: NativeStackNavigationProp<StackParamList, 'Main'>;
+}
+
+const MainScreen = ({navigation}: MainScreenProps): JSX.Element => {
+  return (
+    <ScrollView>
+      <Button
+        onPress={() => navigation.navigate('Settings')}
+        title="Go to next screen"
+      />
+      <Button onPress={() => navigation.pop()} title="🔙 Back to Examples" />
+    </ScrollView>
+  );
+};
+
+interface SettingsScreenProps {
+  navigation: NativeStackNavigationProp<StackParamList, 'Settings'>;
+}
+
+const SettingsScreen = ({navigation}: SettingsScreenProps): JSX.Element => {
+  const [headerTitle, setHeaderTitle] = useState('Settings');
+  const [backButtonVisible, setBackButtonVisible] = useState(true);
+  const [headerShown, setHeaderShown] = useState(true);
+  const [headerLargeTitle, setHeaderLargeTitle] = useState(false);
+  const [headerItem, setHeaderItem] = useState('right');
+  const [headerBackTitle, setHeaderBackTitle] = useState('Back');
+  const [headerHideShadow, setHeaderHideShadow] = useState(false);
+  const [headerTranslucent, setHeaderTranslucent] = useState(false);
+
+  const square = (props: {tintColor?: string}) => (
+    <Square {...props} color="green" size={20} />
+  );
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerTitle,
+      headerBackTitleVisible: backButtonVisible, // iOS
+      headerHideBackButton: !backButtonVisible, // android
+      headerShown,
+      headerRight: headerItem === 'right' ? square : undefined,
+      headerCenter: headerItem === 'center' ? square : undefined,
+      headerLeft: headerItem === 'left' ? square : undefined,
+      headerLargeTitle, // iOS
+      headerBackTitle, // iOS
```

Comment:
Nit; we don't sort this alphabetically anyway, so we can at least group iOS props together.